pyfolio/capacity.py

```diff
@@ -0,0 +1,201 @@
+
+    Returns
+    -------
+    constraint_tickers : pd.DataFrame
+        Algo capacity and constraining ticker at various parts of
+        daily volume distribution.
+    """
+
+    constraint_tickers = pd.DataFrame()
+    for name, d in constraints.dropna(axis=0).iteritems():
+        constraint_tickers.loc[name, 'Algo Capacity $ Millions'] = d.min()
+        constraint_tickers.loc[name, 'Constraining Ticker'] = d.argmin()
+
+    return constraint_tickers
+
+
+def daily_txns_with_bar_data(transactions, market_data):
+    """
+    Sums the absolute value of shares traded in each name on each day.
+    Adds columns containing the closing price and total daily volume for
+    each day-ticker combination.
+
+    Parameters
+    ----------
+    transactions : pd.DataFrame
+        Prices and amounts of executed trades. One row per trade.
+        - See full explanation in tears.create_full_tear_sheet
+    market_data : pd.DataFrame
+
+    Returns
+    -------
+    txn_daily : pd.DataFrame
+        Daily totals for transacted shares in each traded name.
+        price and volume columns for close price and daily volume for
+        the corresponding ticker, respectively.
+    """
+    transactions.index.name = 'date'
+    txn_daily = pd.DataFrame(transactions.assign(
+        amount=abs(transactions.amount)).groupby(
+        ['symbol', pd.TimeGrouper('D')]).sum()['amount'])
+    txn_daily['price'] = market_data['price'].unstack()
+    txn_daily['volume'] = market_data['volume'].unstack()
+
+    txn_daily = txn_daily.reset_index().set_index('date')
+
+    return txn_daily
+
+
+def apply_slippage_penalty(returns, txn_daily, simulate_starting_capital,
+                           backtest_starting_capital, impact=0.1):
+    """
+    Applies quadratic volumeshare slippage model to daily return based
+    on the proportion of the observed historical daily bar dollar volume
+    consumed by the strategy's trades. Scales the size of trades based
+    on the ratio of the starting capital we wish to test to the starting
+    capital of the passed backtest data.
+
+    Parameters
+    ----------
+    returns : pd.Series
+        Time series of daily returns.
+    txn_daily : pd.Series
+        Daily transaciton totals, closing price, and daily volume for
+        each traded name. See price_volume_daily_txns for more details.
+    simulate_starting_capital : integer
+        capital at which we want to test
+    backtest_starting_capital: capital base at which backtest was
+        origionally run. impact: See Zipline volumeshare slippage model
+
+    Returns
+    -------
+    adj_returns : pd.Series
+        Slippage penalty adjusted daily returns.
+    """
+    mult = simulate_starting_capital / backtest_starting_capital
+    simulate_traded_shares = abs(mult * txn_daily.amount)
+    simulate_traded_dollars = txn_daily.price * simulate_traded_shares
+
+    penalties = (simulate_traded_shares / txn_daily.volume)**2 \
```

Comment:
Would be even clearer with a variable in-between: `simulated_pct_volume_used = simulate_traded_shares / txn_daily.volume`.